### PR TITLE
Added the french string for bad hard coded traduction and enhanced us…

### PR DESCRIPTION
**Pull request by samymessouak**
#### Description:
This PR introduces the following updates and improvements to the application:

1. **French Translations**:
   - Added French translations for all hardcoded strings in the XML files.
   - Ensured proper localization by replacing hardcoded strings in the code with references to `strings.xml`.
   - Verified that the application switches correctly between languages based on the device's locale.

2. **Round Tracking in Ongoing Challenges**:
   - Introduced a new field to track the number of rounds completed in the `OngoingChallengeCard`.
   - Displayed the current round progress (e.g., "Round 1/3") directly on the card for better user visibility.
   - Tested the implementation thoroughly to ensure that the round count updates correctly during gameplay and reflects real-time progress.

#### Changes Made:
- **Localization**:
  - Updated `strings.xml` with all the required strings.
  - Added `strings-fr.xml` for French translations.
  - Replaced hardcoded strings in the application with appropriate resource references.
  
- **Ongoing Challenge Updates**:
  - Updated the UI of `OngoingChallengeCard` to include a round progress field.
  - Added logic to dynamically update the round count based on gameplay progress.

#### Testing:
- Verified the correctness of the French translations by testing on a device with the locale set to French.
- Ensured that all UI elements display the appropriate localized strings in both English and French.
- Tested the round progress field to confirm:
  - It displays the correct round number.
  - It remains consistent across different challenge modes.

#### Impact:
- Improves the application's accessibility for French-speaking users.
- Enhances the user experience by providing better visibility into challenge progress.

#### Screenshots:
image

### CODE REVIEW

File: app/src/main/java/com/github/se/signify/ui/screens/challenge/ChallengeHistoryScreen.kt [hunk 1]
```diff
@@ -213,32 +220,36 @@ fun determineWinner(
...
 
 @Composable
 fun PlayerScoreText(player: String, result: Double, mode: String) {
+  val wordsText = stringResource(R.string.words)
+
   val scoreText =
       when (mode) {
         ChallengeMode.CHRONO.toString() -> "$player Score: $result s"
-        else -> "$player Score: $result words"
+        else -> "$player Score: $result $wordsText"
```
Comment (SaFouNaldo) [inline app/src/main/java/com/github/se/signify/ui/screens/challenge/ChallengeHistoryScreen.kt:252]: There is hardocded string here for score even if it's in french "Score"

File: app/src/main/java/com/github/se/signify/ui/screens/challenge/ChallengeScreen.kt [hunk 2]
```diff
@@ -292,14 +298,17 @@ fun OngoingChallengeCard(
...
                 fontSize = 18.sp,
-                color = MaterialTheme.colorScheme.onSurface,
-                fontWeight = FontWeight.Bold)
+                color = MaterialTheme.colorScheme.onSurface)
             Text(
                 text = "${stringResource(R.string.mode_text)}: ${challenge.mode}",
                 fontSize = 14.sp,
                 color = MaterialTheme.colorScheme.onSurface)
+            Text(
+                text = "Round: $round/3",
```
Comment (SaFouNaldo) [inline app/src/main/java/com/github/se/signify/ui/screens/challenge/ChallengeScreen.kt:309]: Also here Round should not be hardcoded even if in French

**Review by SaFouNaldo — APPROVED** [review]
Good PR, removed practically all the hardocded strings except for two.
You can see below.

File: app/src/main/java/com/github/se/signify/ui/screens/challenge/ChallengeScreen.kt (hunk 2 above)
Comment (samymessouak) [reply app/src/main/java/com/github/se/signify/ui/screens/challenge/ChallengeScreen.kt]: Done, ty

File: app/src/main/java/com/github/se/signify/ui/screens/challenge/ChallengeHistoryScreen.kt (hunk 1 above)
Comment (samymessouak) [reply app/src/main/java/com/github/se/signify/ui/screens/challenge/ChallengeHistoryScreen.kt]: Done, ty

**Pull request merged**